src/PowerShellEditorServices/Hosting/EditorServicesServerFactory.cs

```diff
@@ -0,0 +1,110 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices.Server;
+using PowerShellEditorServices.Logging;
+using Serilog;
+using System;
+using System.IO;
+
+namespace Microsoft.PowerShell.EditorServices.Hosting
+{
+    /// <summary>
+    /// Factory class for hiding dependencies of Editor Services.
+    /// In particular, dependency injection and logging are wrapped by factory methods on this class
+    /// so that the host assembly can construct the LSP and debug servers
+    /// without taking logging or dependency injection dependencies directly.
+    /// </summary>
+    internal class EditorServicesServerFactory
+    {
+        /// <summary>
+        /// Create a new Editor Services factory.
+        /// This method will instantiate logging.
+        /// </summary>
+        /// <param name="logPath">The path of the log file to use.</param>
+        /// <param name="minimumLogLevel">The minimum log level to use.</param>
+        /// <returns></returns>
+        public static EditorServicesServerFactory Create(string logPath, int minimumLogLevel, IObservable<(int logLevel, string message)> hostLogger)
+        {
+            Log.Logger = new LoggerConfiguration()
+                .Enrich.FromLogContext()
+                .WriteTo.File(logPath)
+                .MinimumLevel.Verbose()
+                .CreateLogger();
+
+            ILoggerFactory loggerFactory = new LoggerFactory().AddSerilog(Log.Logger);
+
+            // Hook up logging from the host so that its recorded in the log file
+            hostLogger.Subscribe(new HostLoggerAdapter(loggerFactory));
+
+            return new EditorServicesServerFactory(loggerFactory, (LogLevel)minimumLogLevel);
+        }
+
+        private readonly ILoggerFactory _loggerFactory;
+
+        private readonly Extensions.Logging.ILogger _logger;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unread private field '_logger' or refactor the code to use its value.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)